pkg/services/sqldb/provision.go

```diff
@@ -95,16 +176,96 @@ func (s *serviceManager) preProvision(
 	return dt, nil
 }
 
-func buildARMTemplateParameters(
+func (v *vmOnlyManager) preProvision(
+	_ context.Context,
+	instance service.Instance,
+	_ service.Plan,
+	_ service.Instance, //reference instance
+) (service.InstanceDetails, error) {
+	dt, ok := instance.Details.(*mssqlVMOnlyInstanceDetails)
+	if !ok {
+		return nil, errors.New(
+			"error casting instance.Details as *mssqlVMOnlyInstanceDetails",
+		)
+	}
+	dt.ARMDeploymentName = uuid.NewV4().String()
+	dt.ServerName = uuid.NewV4().String()
+	dt.AdministratorLogin = generate.NewIdentifier()
+	dt.AdministratorLoginPassword = generate.NewPassword()
+	return dt, nil
+}
+
+func (d *dbOnlyManager) preProvision(
+	_ context.Context,
+	instance service.Instance,
+	_ service.Plan,
+	referenceInstance service.Instance, //reference instance
+) (service.InstanceDetails, error) {
+	dt, ok := instance.Details.(*mssqlInstanceDetails)
+	if !ok {
+		return nil, errors.New(
+			"error casting instance.Details as *mssqlInstanceDetails",
+		)
+	}
+	//Assume refererence instance is a vm only instance?
```

Comment:
Yes. Good assumption. We should be failing if the reference instance isn't the kind of instance we expected it to be.